### DISCUSSION TITLE
feedback

### DISCUSSION THREAD

**Discussion by giscus** [mrocklin/mrocklin.github.io#18] [Giscus]
# feedback

Feedback is critical to performing good work as a team. Good feedback cultivates quality work and professional growth. Bad feedback degrades quality and erodes relationships. This article explains ...

https://matthewrocklin.com/feedback.html

**Comment by avi-newman-seek** [comment]
Thanks Matt!

I really appreciate the spirit behind this post and how difficult it is to give constructive feedback because many people have insecurities that make it difficult to receive feedback. As you say, context is everything. "I'm investing in you" vs "Your ignorance is holding us all back".  The only thing I see missing here is a reminder to also give positive feedback in other parts of the review where applicable. Even basic feedback like, "I can see you are trying..." or "I can see your good intentions, keep it up!"

Some folks believe that giving negative feedback just isn't possible, and that it's best to ONLY give positive feedback. This means trying to steer folks towards doing what they are naturally good at, but it does mean giving up on helping the person to be more than what they are. 

What's painful for me, is that even though I feel like I'm doing someone a favor by spending significant effort to invest in them, it feels like I also need to pay them with compliments and other social investments to earn their good will and attention to inoculate their ego. As someone who has a lot of insights into people's limitations, I have to constantly remind myself to pick my battles and remember that Sun Tzu taught that **the ultimate teacher is consequences**. If someone can learn by failing in a safe way, that's going to be far better more effective for them than anything I can say to them.

**Reply by mrocklin** [reply]
Thanks Avi, well said.

**Comment by bryan-lott** [comment]
Great article. One thing that struck me and wasn't called out explicitly is there's a power dynamic at play as well in the Corporate world that is drastically different than the OSS world. In OSS, as you said, everyone is a volunteer. The power dynamics at play, I suspect, are largely based on skill, talent, and "I got there first"ism. At worst, if a reviewee gets "fired", they can no longer contribute to the project. 
That's a very different dynamic than "this person that's reviewing me can fire me and impact my ability to provide for myself and my family." Criticism in the corporate world can come across as potentially "life threatening", depending on the person's circumstances.

**Comment by rogerrohrbach** [comment]
I continuously send the Open Source message as a technical lead in a corporate setting.  I'd not characterize this as "Open Source," but rather, "Constructive," or "Talent-Nurturing."

OTOH I'm ashamed to say I've repeatedly sent the Corporate message to a developer who evinces no interest in improving their craft.  This is obviously a failing strategy; I keep hoping the individual will feel some shame (as I would in their place) and change their behavior.  In an ideal world, I'd not retain this developer, but in a corporate setting it sometimes transpires that there is work to be done that only such a person will deign to do...

**Comment by benjiwheeler** [comment]
I agree that receiving the corporate example of feedback would feel lousy. But that type of feedback is basically necessary, right? How do we make it better?
